src/backend/distributed/commands/dependencies.c

```diff
@@ -153,8 +153,15 @@ EnsureDependenciesCanBeDistributed(const ObjectAddress *objectAddress)
 	/* If the object circularcly depends to itself, Citus can not handle it */
 	ErrorIfCircularDependencyExists(objectAddress);
 
-	/* If the object has any unsupported dependency, error out */
-	DeferredErrorMessage *depError = DeferErrorIfHasUnsupportedDependency(objectAddress);
+	/*
+	 * If the object has any unsupported dependency, error out.
+	 *
+	 * Also, we will abort the transaction on any error. So need to set
+	 * createInProgress to false.
+	 */
+	bool createInProgress = false;
```

Comment:
Can we drop `createInProgress` flag altogether, and here when `if (depError != NULL)`, we override the detail to be empty string? 

It is because:
a) we'd not need `createInProgress=false` anywhere else, and future callers wouldn't need to think about what value to pass for this flag
b) The name of `createInProgress` doesn't really clearly explain what it does in the function. And, finding a good name for this seems non-trivial.  

```C
/* override DETAIL as it is not needed here*/
 DeferredError(depError->code,
						 depError->message,
						 NULL /* no detail*/,
						 depError->hint);
```